src/main/java/ui/TestController.java

```diff
@@ -157,4 +159,30 @@ public static RepoIO createTestingRepoIO(Optional<JSONStore> jsonStoreToBeUsed)
         return new RepoIO(Optional.of(new DummySource()), jsonStoreToBeUsed,
                           Optional.of(RepoStore.TEST_DIRECTORY));
     }
+
+    /**
+     * Create an update progress window if not in test mode.
+     *
+     * @return update progress window
+     */
+    public static UpdateProgressWindow createUpdateProgressWindow() {
+        if (isTestMode()) {
+            return null;
```

Comment:
I wanted to create stub for this but javafx stage initialization happens in the constructor which can't be changed in the stub, hence I use `null`. Do you think this is fine?

A problem I encounter is that some guitests seems to be bypassing `isTestMode()` which ends up calling update progress window and giving error.